opencaesar/oml-vision
# Add Custom Commands in Context Menu

## Checklist before submitting a merge request


## Description of contribution

This feature allows users to add custom commands (SPARQL queries) which are displayed and selected to be run through a context menu.

closes #20 

## Testing performed



- [x] I have added unit tests to cover additional functionality (backend only, frontend tests still need to be added)

## How to Test Expected functionality changes

1. Go to https://github.com/pogi7/kepler16b-example
2. Switch to `main` branch
3. From the command line run `./gradlew load`
4. Wait for that command to finish
5. Open the `Objectives` webview
6. Click on any row to select the row then right click the selected row
7. Click `Create Objective`
8. You should see the table update
9. From the command line run `./gradlew save`
10. You should see the `objectives.oml` file change.  You can run `git status` to verify.
11. Click on any row to select the row then right click the selected row
12. Click `Delete Objective`
13. You should see the table update
14. From the command line run `./gradlew save`
15. You should see the changes in the `objectives.oml` file go away.  You can run `git status` to verify.
16. Open the `Components` webview
17. Click on any node to select the component then right click the selected component
18. Click `Create Component`
19. You should see the diagram update
20. From the command line run `./gradlew save`
21. You should see the `components.oml` file change.  You can run `git status` to verify.
22. Click on any row to select the row then right click the selected row
23. Click `Delete Component`
24. You should see the component update
25. From the command line run `./gradlew save`
26. You should see the changes in the `components.oml` file go away.  You can run `git status` to verify.

## Additional context

[If needed, you may add additional context]
